File: Laravel/proveEsami/careerExams/app/Http/Controllers/ExamController.php
<?php

namespace App\Http\Controllers;

use App\Models\Exam;
use App\Models\Career;

use Illuminate\Http\Request;

class ExamController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        $exams = Exam::all();
        return view('exam.index', compact('exams'));
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        $careers = Career::all();
        return view('exam.create', compact('careers'));

    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Exam::create($request->all());
        return redirect("/exams");
    }

    /**
     * Display the specified resource.
     */
    public function show(Exam $exam)
    {
        $careers = Career::all();
        return view('exam.show', compact('exam', 'careers'));
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Exam $exam)
    {
        //
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Exam $exam)
    {
        $exam->update($request->all());
        return $this->show($exam);
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Exam $exam)
    {
        $exam->delete();
        return redirect("/exams");
    }
    public function deleteAll()
    {
        $exams = Exam::all();
        foreach($exams as $exam){
            $exam->delete();
        }
        return redirect("/exams");
    }
}

File: SpringBoot/proveEsame/career_exams/src/main/java/com/example/career_exams/controllers/ExamController.java
package com.example.career_exams.controllers;

import java.util.List;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;

import com.example.career_exams.data.CareerRepository;
import com.example.career_exams.data.ExamRepository;
import com.example.career_exams.models.Career;
import com.example.career_exams.models.Exam;


import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;


@RequestMapping("/exams")
@Controller
public class ExamController {

    private ExamRepository examRepository;
    private CareerRepository careerRepository;

    public ExamController(CareerRepository careerRepository, ExamRepository examRepository){
        this.careerRepository = careerRepository;
        this.examRepository = examRepository;
    }

    @GetMapping
    public String index(Model model) {
        List<Exam> exams = examRepository.findAll();
        model.addAttribute("exams", exams);
        return "exam/index";
    }

    @GetMapping("/create")
    public String create(Model model) {
        List<Career> careers = careerRepository.findAll();
        model.addAttribute("careers", careers);
        return "exam/create";
    }

    @GetMapping("/{id}")
    public String show(@PathVariable Long id, Model model) {
        List<Career> careers = careerRepository.findAll();
        model.addAttribute("careers", careers);
        Exam exam = examRepository.findById(id).orElse(null);
        model.addAttribute("exam", exam);
        return "exam/show";
    }


    @PostMapping("/{id}")
    public String update(Exam exam) {
        examRepository.save(exam);
        return "redirect:/exams/"+exam.getId();
    }

    @PostMapping("/delete/{id}")
    public String delete(@PathVariable Long id) {
        examRepository.deleteById(id);
        return "redirect:/exams";
    }

    @PostMapping
    public String store(Exam exam) {
        examRepository.save(exam);
        return "redirect:/exams";
    }

    @GetMapping("/deleteAll")
    public String deleteAllExams() {
        List<Exam> exams = examRepository.findAll();
        for (Exam exam : exams) {
            examRepository.delete(exam);
        }
        return "redirect:/exams";
    }
    
}
